### ISSUE TITLE
Detect loss function's `reduction`, error if unsupported

### ISSUE THREAD

**Issue by f-dangel** [f-dangel/vivit#22]
For now, the library only supports `reduction='mean'`. We rely on the user to use this reduction and raise awareness about this point in the documentation. It would be better to automatically have the library detect the reduction and error if it is unsupported.

This can be done via a hook into BackPACK.
- [ ] Implement hook that determines the loss function reduction during backpropagation
- [ ] Integrate the above hook into the `*Computation` and raise an exception if the reduction is not supported
- [ ] Remove the comments about supported reductions in the documentation
